### ISSUE TITLE
Error when using atom-package-deps to automatically install atom-ide-ui

### ISSUE THREAD

**Issue by henryju** [atom/atom-languageclient#191]
Hi,

Since my package is relying on atom-ide-ui to show diagnostics and code actions, I gave a try to *atom-package-deps* to automatically trigger its installation.
The code looks like:
```javascript
class SonarLintLanguageClient extends AutoLanguageClient {
  async activate(state) {
    const pns = atom.packages.getAvailablePackageNames()
    if (!(pns.includes("atom-ide-ui") || pns.includes("linter"))) {
      await require("atom-package-deps").install("sonarlint", true);
    }
    super.activate(state);
  [...]
}
```

When starting Atom with my package but not linter nor atom-ide-ui, the popup is correctly displayed asking to install atom-ide-ui. But if I click "yes", there is this error:

```
Failed to activate the atom-ide-diagnostics package
Cannot read property 'getActiveServers' of undefined
TypeError: Cannot read property 'getActiveServers' of undefined
    at SonarLintLanguageClient.consumeLinterV2 (/home/julien/tmp/sonarlint/node_modules/atom-languageclient/build/lib/auto-languageclient.js:435:49)
    at Provider.module.exports.Provider.provide (/usr/share/atom/resources/app.asar/node_modules/service-hub/lib/provider.js:34:52)
    at ServiceHub.module.exports.ServiceHub.provide (/usr/share/atom/resources/app.asar/node_modules/service-hub/lib/service-hub.js:30:20)
    at Package.activateServices (/usr/share/atom/resources/app.asar/src/package.js:392:69)
    at Package.activateNow (/usr/share/atom/resources/app.asar/src/package.js:228:14)
    at measure (/usr/share/atom/resources/app.asar/src/package.js:201:27)
    at Package.measure (/usr/share/atom/resources/app.asar/src/package.js:83:19)
    at activationPromise.Promise (/usr/share/atom/resources/app.asar/src/package.js:195:14)
    at Package.activate (/usr/share/atom/resources/app.asar/src/package.js:193:32)
    at PackageManager.activatePackage (/usr/share/atom/resources/app.asar/src/package-manager.js:690:36)
    at _features.forEach.feature (/home/julien/.atom/packages/atom-ide-ui/modules/nuclide-commons-atom/FeatureLoader.js:208:23)
    at Array.forEach (native)
    at FeatureLoader.activate (/home/julien/.atom/packages/atom-ide-ui/modules/nuclide-commons-atom/FeatureLoader.js:206:20)
    at Object.activate (/home/julien/.atom/packages/atom-ide-ui/index.js:78:19)
    at Package.activateNow (/usr/share/atom/resources/app.asar/src/package.js:225:27)
    at measure (/usr/share/atom/resources/app.asar/src/package.js:201:27)
    at Package.measure (/usr/share/atom/resources/app.asar/src/package.js:83:19)
    at activationPromise.Promise (/usr/share/atom/resources/app.asar/src/package.js:195:14)
    at Package.activate (/usr/share/atom/resources/app.asar/src/package.js:193:32)
    at PackageManager.activatePackage (/usr/share/atom/resources/app.asar/src/package-manager.js:690:36)
    at /home/julien/.atom/packages/linter-ui-default/node_modules/atom-package-deps/lib/index.js:36:35
    at Generator.next (<anonymous>)
    at step (/home/julien/.atom/packages/linter-ui-default/node_modules/atom-package-deps/lib/index.js:66:191)
    at /home/julien/.atom/packages/linter-ui-default/node_modules/atom-package-deps/lib/index.js:66:361
The error was thrown from the atom-ide-diagnostics package.
```

In the console, there is only `GET https://atom.io/api/packages/atom-ide-diagnostics 404 (Not Found)` that come from the notification manager. 

I'm suspecting a problem in package activation order, but I'm not enough experimented with the package / service architecture to find the root cause.

Any idea?

**Comment by laughedelic** [comment]
Probably you shouldn't check manually `pns.includes("atom-ide-ui") || pns.includes("linter")`, because atom-package-deps will do it anyway? 
I'm [using it](https://github.com/laughedelic/atom-ide-scala/blob/d4716b0044f10c45382b942ce31c111ea46d8a75/src/main/scala/Exports.scala#L17) without problems to install `atom-ide-ui` and `language-scala` dependencies. And I think it's not even necessary to wait for the promise..

**Comment by henryju** [comment]
The check is necessary because linter and atom-ide-ui (as well as nuclide that I added later) are concurrent implementation of the diagnostics feature. If users already have nuclide installed, I don't want to prompt them to install atom-ide-ui.

But NOT waiting for the promise seems to indeed fix my issue. Thanks!

Fixed code: https://github.com/SonarSource/sonarlint-atom/blob/4be0378a3f5b362ad47590357fc03d5a9261a5e0/lib/sonarlint.js#L47
